### ISSUE TITLE
Add Grunt and NPM to manage dependencies

### ISSUE THREAD

**Issue by danjford** [s16h/py-must-watch#73]
Can we add e.g. Grunt to run tasks such as minifying and NPM to manage dependencies?

**Comment by techtonik** [comment]
@danjford do you have a tutorial explaining what all those is about for Python?

**Comment by danjford** [comment]
It wouldn't really be for the python, it is more a better way to automate and manage the front end code and build process for the site.

**Comment by techtonik** [comment]
Why not use Python toolchain for that?

**Comment by s16h** [comment]
Great suggestion. The website and its dependencies are very simple and slow-changing currently. So, I don't think we need tools. However, I can most certainly see a future where this would be necessary.

**Comment by techtonik** [comment]
Can anybody point to a tutorial about this Grunt/Gulp etc. automation?

**Comment by danjford** [comment]
@techtonik [This](http://www.smashingmagazine.com/2013/10/get-up-running-grunt/) may be a good place to start.

**Comment by techtonik** [comment]
@danjford looks like Grunt is just [fabric](http://www.fabfile.org/), but Grunt files don't look readable to me.

Have you seen http://blog.keithcirkel.co.uk/why-we-should-stop-using-grunt/ and http://brunch.io/ - they say that Grunt/Gulp is an overkill.

**Comment by danjford** [comment]
I don't think fabric and Grunt are that similar, perhaps it's because you aren't that familiar with Grunt?

I have seen their posts, however for every library there will always be a post against it. But at the same time there are going to be posts elsewhere which are for it. As for brunch.io, they are going to be against Grunt as they are a competitor.

If you want to have a look, @techtonik  and @s16h, I've done an implementation of py-must-watch with npm and grunt.

https://github.com/danjford/py-must-watch/tree/development

**Comment by nloadholtes** [comment]
@techtonik Thanks for those links, as a Python dev the whole JS ecosystem is a mystery to me. That first link really helped me understand what all is going on.

**Comment by techtonik** [comment]
@nloadholtes Python ecosystem also looks strange from outside. I know people who think that `setup.py` is just weird. But the question about JS ecosystem is important for maintainability. I understand that SASS and Coffeescript need to be compiled and minified, but when I tried to understand what all this stuff for https://grtp.co/ I realized that JS ecosystem is beyond my understanding. Now it is more clear thanks to @danjford mentioning it again.
